pkgs/development/python-modules/cryptography/default.nix

```
@@ -69,6 +75,11 @@ buildPythonPackage rec {
     pytz
   ];
 
+  CARGO_BUILD_TARGET = "${rust.toRustTargetSpec stdenv.hostPlatform}";
+  PYO3_CROSS_LIB_DIR = "${python}/lib/${python.libPrefix}";
+  "CARGO_TARGET_${lib.toUpper (builtins.replaceStrings ["-"] ["_"] (rust.toRustTarget stdenv.hostPlatform))}_LINKER" =
+    "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
```

```suggestion
  CARGO_BUILD_TARGET = rust.toRustTargetSpec stdenv.hostPlatform;
  PYO3_CROSS_LIB_DIR = "${python}/lib/${python.libPrefix}";
  "CARGO_TARGET_${lib.toUpper (builtins.replaceStrings ["-"] ["_"] (rust.toRustTarget stdenv.hostPlatform))}_LINKER" =
    "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
```

Are those generic variables we want to set?
If so we should move them to a setup-hook like maturin or setuptools-rust

---

That is the question... I took those variables from here https://github.com/PyO3/setuptools-rust/blob/main/.github/workflows/ci.yml#L270 and here https://github.com/PyO3/setuptools-rust/issues/294#issuecomment-1264326401. The `CARGO_BUILD_TARGET` and `PYO3_CROSS_LIB_DIR` seem reasonable, but I am not sure about `CARGO_TARGET_*`. It is kind of weird to have to specify the linker. No other package except of `crosvm` has to do this (that is where I took it from anyway).

---

> It is kind of weird to have to specify the linker.

Especially weird given that we already do this: https://github.com/NixOS/nixpkgs/blob/06542b21a03083b894c0b87486aa67b65280a028/pkgs/build-support/rust/hooks/default.nix#L98-L110